src/rust/mangastream/template/src/template.rs

```diff
@@ -331,18 +369,36 @@ impl MangaStreamSource {
 
 				title.join(" ")
 			};
-			let chapter_url = {
+
+			let chapter_url: String;
+			let chapter_id: String;
+
+			if self.use_postids {
 				let original_url = chapter_node.select(self.chapter_url).attr("href").read();
 
-				if self.has_permanent_chapter_url {
-					get_permanet_url(original_url)
-				} else {
-					original_url
-				}
-			};
-			let chapter_id = get_id_from_url(chapter_url.clone());
+				chapter_id = String::from(
+					chapter_url_to_postid_mapping
+						.get(&original_url)
+						.expect("Failed to get chapter id from url mapping"),
```

Comment:
probably just `?` here

Reply:
Had to have an `AidokuError` and none of them seemed appropriate, so I just went with `AidokuErrorKind::Unimplemented`